dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/MetadataBitstreamRestRepository.java

```diff
@@ -347,28 +352,22 @@ public String extractFile(InputStream inputStream, String fileType) {
 
         StringBuilder sb = new StringBuilder();
         sb.append(("<root>"));
-        List<String> allFiles = filePaths;
+        Iterator<String> iterator = filePaths.iterator();
         int fileCounter = 0;
-        for (String filePath : allFiles) {
-            if (!filePath.isEmpty() && filePath.length() > 3) {
+        while ((iterator.hasNext() && fileCounter < maxPreviewCount)) {
+            String filePath = iterator.next();
+
+            if (filePath.length() > 3) {
                 if (filePath.contains(".")) {
                     fileCounter++;
                 }
-                sb.append("<element>");
-                sb.append(filePath);
-                sb.append("</element>");
-
-                if (fileCounter > MAX_FILE_PREVIEW_COUNT) {
-                    sb.append("<element>");
-                    sb.append("/|0");
-                    sb.append("</element>");
-                    sb.append("<element>");
-                    sb.append("...too many files...|0");
-                    sb.append("</element>");
-                    break;
-                }
+                sb.append("<element>").append(filePath).append("</element>");
             }
         }
+
+        if (fileCounter > maxPreviewCount) {
+            sb.append("<element>...too many files...|0</element>");
```

Comment:
what is this `|0` at the end? seems out of place

Reply:
Yes, but that `|0` has a higher purpose, that value is converted to Bytes info in the FE.
The result looks like this: 
![image](https://github.com/dataquest-dev/DSpace/assets/90026355/e857893a-dbd8-4950-adbb-16ac8e627743)

The image is created in the v5

Reply:
that is absolutely awful.. why is it in its own folder?
and it is not really true it has 0 bytes, is it :) 

But if you are ok with it, I don't want to postpone this even more. 
But I DO NOT like it.